arch/risc-v/src/common/riscv_swint.c

```diff
@@ -223,6 +223,16 @@ int riscv_swint(int irq, void *context, void *arg)
         {
           DEBUGASSERT(regs[REG_A1] != 0);
           CURRENT_REGS = (uintptr_t *)regs[REG_A1];
+#ifdef CONFIG_ARCH_ADDRENV
+
+          /* Make sure that the address environment for the previously
+           * running task is closed down gracefully (data caches dump,
+           * MMU flushed) and set up the address environment for the
+           * new thread at the head of the ready-to-run list.
+           */
+
+          group_addrenv(NULL);
```

Comment:
don't need? done in up_doirq.

Reply:
up_doirq for risc-v is not actually in use, so in this patch moved these function call into swint.c to avoid duplicated code in each chip.

We can move it to up_doirq until it can be used, but need further work (unify the plic/clic definition across chips).

Reply:
Now it handle by riscv_doirq on all riscv based chips.